.github/workflows/main.yml

```diff
@@ -40,25 +64,242 @@ jobs:
+          if [ -z "$ENVIRONMENT_UID" ] || [ "$ENVIRONMENT_UID" = "null" ]; then
+            echo "No Postman environment named '$POSTMAN_ENV_NAME' found in workspace."
+            exit 1
+          fi
+          
+          ENV_COUNT=$(echo "$ENVS_JSON" | jq -r --arg n "$POSTMAN_ENV_NAME" '[.environments[] | select(.name==$n)] | length')
+          if [ "$ENV_COUNT" -gt 1 ]; then
+            echo "Warning: Found $ENV_COUNT environments named '$POSTMAN_ENV_NAME'. Using the most recently updated one."
+          fi
+
+          curl -s -H "X-Api-Key: ${POSTMAN_API_KEY}" \
+            "https://api.getpostman.com/environments/${ENVIRONMENT_UID}" \
+            -o "postman/${TARGET_ENV}.environment.json"
+
+          # 5) Fetch Postman globals if available
+          echo "Fetching Postman globals..."
+          GLOBALS_JSON=$(curl -s -H "X-Api-Key: ${POSTMAN_API_KEY}" \
+            "https://api.getpostman.com/workspaces/${POSTMAN_WORKSPACE_ID}")
+          
+          GLOBALS_UID=$(echo "$GLOBALS_JSON" | jq -r '.workspace.globals.id // empty')
+          
+          if [ -n "$GLOBALS_UID" ]; then
+            curl -s -H "X-Api-Key: ${POSTMAN_API_KEY}" \
+              "https://api.getpostman.com/environments/${GLOBALS_UID}" \
+              -o "postman/globals.json"
+            echo "Globals downloaded"
+            echo "globals_flag=-g postman/globals.json" >> "$GITHUB_OUTPUT"
+          else
+            echo "No globals found in workspace"
+            echo "globals_flag=" >> "$GITHUB_OUTPUT"
+          fi
+
+          echo "env_file=postman/${TARGET_ENV}.environment.json" >> "$GITHUB_OUTPUT"
+
+      - name: Setup Node for Newman
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+          cache: npm
+
+      - name: Install Newman
+        run: npm i -g newman newman-reporter-htmlextra
+
+      - name: Inject sensitive variables (optional)
+        if: always()
+        shell: bash
+        run: |
+          # If you keep sensitive secrets out of Postman, you can override here.
+          # Example: pass API tokens as Newman env-vars (picked in collection via {{apiToken}})
+          case "$TARGET_ENV" in
+            dev)  TOKEN="${{ secrets.DEV_API_TOKEN }}" ;;
+            qa)   TOKEN="${{ secrets.QA_API_TOKEN }}" ;;
+            *)    TOKEN="" ;;
+          esac
+          echo "API_TOKEN=$TOKEN" >> "$GITHUB_ENV"
+
+      - name: Run all Postman collections with Newman
+        shell: bash
+        run: |
+          set -e
+          mkdir -p newman
+          ENV_FILE="${{ steps.fetch-postman.outputs.env_file }}"
+          GLOBALS_FLAG="${{ steps.fetch-postman.outputs.globals_flag }}"
+          # Optionally, run only a "Smoke" folder from each collection first:
+          # FOLDER_FLAG='--folder "Smoke"'
+          FOLDER_FLAG=''
+          FAILED_COLLECTIONS=0
+          TOTAL_COLLECTIONS=0
+          for COL in postman/*.collection.json; do
```

Comment:
It's also odd to have these on disk, when we could just have them in memory, no? They aren't _that_ large, right?